astropy/cosmology/tests/test_connect.py

```diff
@@ -63,31 +75,39 @@ def test_readwrite_complete_info(self, cosmo, tmpdir, format):
         got = Cosmology.read(fname, format=format)
 
         assert got == cosmo
-        assert dict(got.meta) == dict(cosmo.meta)
+        if metaio:
+            assert dict(got.meta) == dict(cosmo.meta)
```

Comment:
I've thought of a better test than the quick workaround suggested here. Please change this throughout this PR.

```suggestion
        assert (not metaio) ^ (dict(got.meta) == dict(cosmo.meta))
```

This should have the truth table

| I/O ↓ equal → |True | False |
| --- | --- | --- |
| **True** | True | False |
| **False** | False | True |